src/sage/rings/polynomial/polynomial_element_generic.py

```diff
@@ -403,9 +404,10 @@ def _repr(self, name=None):
                 else:
                     var = ""
                 s += "%s%s" % (x,var)
-        s = s.replace(" + -", " - ")
-        s = s.replace(" 1*"," ")
-        s = s.replace(" -1*", " -")
+        if not isinstance(self.parent().base(), TropicalSemiring):
```

Comment:
I am -1 on this. The standard printing is how most people write them in papers as I understand it.

Reply:
From a few references, it looks like that they do not use standard printing:
![image](https://github.com/sagemath/sage/assets/116792641/128e9823-1013-460b-9fd0-85f30570f89a)

In others,
![image](https://github.com/sagemath/sage/assets/116792641/46b5036b-2a52-4aeb-9640-e75b5e7bb857)

The terms always separated by `+` because there is no subtraction in tropical polynomials. If the coefficient is 1, it is written so to not confuse it with coefficient 0.

Reply:
If you really want this, then you should override it in your class.